### PR TITLE
Creation of ELK plugin

### PR DESCRIPTION
**Pull request by mmoreram**
This is a complex PR.

- Some of the code inside the folder of Domain was actually requiring external libraries, and this should never be an option. In this case, we have created some interfaces in order to invert the dependencies (for example, a class to enqueue domain events)
- We are basically building a plugin-based project, where we could replace all external communication part as easy as we could
    - ELK plugin - Messages are sent to a redis queue. Basic Infrastructure will start distribute both kibana and logstash
    - RSQueue plugin - Messages are produced and consumed by using Redis as a simple queue
    - RedisStorage - Tokens are stored inside Redis
- Events and Logs repository can be deleted from the PHP client

### CODE REVIEW

**Comment by mmoreram** [comment]
@hkulekci can you please check this PR?
I know, some big changes, but interesting ones if we want Apisearch to be an agnostic project, not coupled with anything :)

File: Plugin/Elastica/Domain/ElasticaWrapper.php [hunk 1]
```diff
@@ -409,7 +409,7 @@ public function addDocuments(
         array $documents
     ) {
         try {
-            $this
+            $r = $this
```
Comment (hkulekci) [inline Plugin/Elastica/Domain/ElasticaWrapper.php]: No need this assignment.

Comment (mmoreram) [reply Plugin/Elastica/Domain/ElasticaWrapper.php]: Fixed

**Comment by mmoreram** [comment]
@hkulekci this PR will need some changes on other libraries and docs.
Ping me as soon as you've checked :)

**Pull request merged**